modules/redis/options.go

```diff
@@ -0,0 +1,141 @@
+package redis
+
+import (
+	"crypto/tls"
+	"net"
+	"strconv"
+
+	"github.com/mdelapenya/tlscert"
+
+	"github.com/testcontainers/testcontainers-go"
+)
+
+type options struct {
+	tlsPort          string
+	withSecureURL    bool
+	withMTLSDisabled bool
+	tlsConfig        *tls.Config
+}
+
+// Compiler check to ensure that Option implements the testcontainers.ContainerCustomizer interface.
+var _ testcontainers.ContainerCustomizer = (Option)(nil)
+
+// Option is an option for the Redpanda container.
+type Option func(*options) error
+
+// Customize is a NOOP. It's defined to satisfy the testcontainers.ContainerCustomizer interface.
+func (o Option) Customize(*testcontainers.GenericContainerRequest) error {
+	// NOOP to satisfy interface.
+	return nil
+}
+
+// WithTLS sets the TLS configuration for the redis container, setting
+// the port to listen on for TLS connections, whether to use a secure URL,
+// and whether to disable MTLS (mutual TLS).
+func WithTLS(tlsPort string, secureURL bool, mtlsDisabled bool) Option {
```

Comment:
Question: do we want three separate options or just this one?

Reply:
Let's be opinionated and not allow passing a port, let the module use a default one.

Same for the the schema, and the mtls, so we could have just WithTLS(), because advance users can always replace the CMD with WithCmd or WithEntrypoint